File: Java/app/src/main/java/vdo/ai/sample/java/core/RewardedVideoActivity.java
package vdo.ai.sample.java.core;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import vdo.ai.android.core.VdoRewardedVideoAd;
import vdo.ai.android.core.listeners.VdoRewardedListener;
import vdo.ai.android.core.models.VdoAdError;
import vdo.ai.sample.java.core.databinding.ActivityRewardedVideoBinding;

public class RewardedVideoActivity extends AppCompatActivity {

    ActivityRewardedVideoBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityRewardedVideoBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        example();

    }

    public void example() {


        VdoRewardedVideoAd rewardedVideoAd = new VdoRewardedVideoAd.VdoRewardedVideoAdBuilder()
                .withContext(this)
                .setEnvironment(BuildConfig.BUILD_TYPE)
                .setTagName("in-app-sample-rw-Z1")
                .setAllowRefresh(false)
                .setListener(new VdoRewardedListener() {

                    @Override
                    public void onUserEarnedReward(int i, @NonNull String s) {
                        binding.title.setText("Reward Amount " + i + ", type: " + s);
                    }

                    @Override
                    public void onAdShowedFullScreenContent() {

                    }

                    @Override
                    public void onAdLoaded() {

                    }

                    @Override
                    public void onAdImpression() {

                    }

                    @Override
                    public void onAdFailedToShowFullScreenContent(@Nullable VdoAdError vdoAdError) {

                    }

                    @Override
                    public void onAdFailedToLoad(@Nullable VdoAdError vdoAdError) {

                    }

                    @Override
                    public void onAdDismissedFullScreenContent() {

                    }

                    @Override
                    public void onAdClicked() {

                    }
                }).build();
    }
}

File: Kotlin/core/src/main/java/vdo/ai/sample/kotlin/RewardedVideoActivity.kt
package vdo.ai.sample.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import vdo.ai.android.core.VdoRewardedVideoAd
import vdo.ai.android.core.listeners.VdoRewardedListener
import vdo.ai.android.core.models.VdoAdError
import vdo.ai.sample.kotlin.databinding.ActivityRewardedVideoBinding


/**
 *  created by Ashish Saini at 5th Oct 2023
 *
 **/
class RewardedVideoActivity : AppCompatActivity() {

    lateinit var  binding: ActivityRewardedVideoBinding
    var TAG= RewardedVideoActivity :: class.java.simpleName
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityRewardedVideoBinding.inflate(layoutInflater)
        setContentView(binding.root)

        example()
    }

    private fun example(){

        val rewardedVideoAd =  VdoRewardedVideoAd.VdoRewardedVideoAdBuilder()
            .withContext(this)
            .setEnvironment(BuildConfig.BUILD_TYPE)
            .setTagName("in-app-sample-rw-Z1")
            .setAllowRefresh(false)
            .setListener( object : VdoRewardedListener {

                override fun onAdLoaded() {
                    Log.d(TAG, "GAM onAdLoaded >>>>>>>>> ")

                }

                override fun onAdImpression() {
                    Log.d(TAG, "GAM onAdImpression >>>>>>>>> ")

                }

                override fun onAdFailedToLoad(adError: VdoAdError?) {
                    Log.d(TAG, "GAM onAdFailedToLoad $adError >>>>>>>>> ")

                }

                override fun onAdClicked() {
                    Log.d(TAG, "GAM onAdClicked >>>>>>>>> ")

                }

                override fun onAdDismissedFullScreenContent() {
                    Log.d(TAG, "GAM onAdDismissedFullScreenContent >>>>>>>>> ")

                }

                override fun onAdFailedToShowFullScreenContent(adError: VdoAdError?) {
                    Log.d(TAG, "GAM onAdFailedToShowFullScreenContent $adError>>>>>>>>> ")

                }

                override fun onAdShowedFullScreenContent() {
                    Log.d(TAG, "GAM onAdShowedFullScreenContent >>>>>>>>> ")

                }

                override fun onUserEarnedReward(amount: Int, type: String) {
                    Log.d(TAG, "GAM onUserEarnedReward amount : $amount >>>>>>>>> ")
                }

            }).build()
    }
}